instance/instance_upgrade.py

```diff
@@ -0,0 +1,106 @@
+            use_ephemeral_databases=False,  # and use persistent databases.
+        )
+
+    def do_upgrade(self, instances):
+        pass
+
+    @staticmethod
+    def _replace_extra_config(instance, target_string, new_value):
+        instance.configuration_extra_settings = instance.configuration_extra_settings.replace(target_string, new_value)
+
+    @staticmethod
+    def clean_up_after_upgrade(instances):
+        """
+        Should be invoked after all instances are updated
+        """
+        pass
+
+    def upgrade_instances(self):
+        instances = self.get_instances_to_upgrade()
+        self.do_upgrade(instances)
+        # TODO: schedule clean_up_after_upgrade after instances are updated (huey task with conditional?)
+
+
+class DogwoodToEucalyptus1(InstanceUpgrade):
+    INITIAL_RELEASE = "dogwood"
+
+    def do_upgrade(self, instances):
+        """Upgrade all OpenEdXInstances in the given iterable to Eucalyptus."""
+        for instance in instances:
+            instance.refresh_from_db()
+            logger.info("Upgrading instance %s to Eucalyptus.1 ...", instance)
+            instance.configuration_extra_settings += "\nCOMMON_EUCALYPTUS_UPGRADE: true\n"
+            instance.edx_platform_repository_url = "https://github.com/open-craft/edx-platform"
+            instance.edx_platform_commit = "opencraft-release/eucalyptus.1"
+            instance.configuration_source_repo_url = "https://github.com/open-craft/configuration"
+            instance.configuration_version = "opencraft-release/eucalyptus.1"
+            instance.openedx_release = "open-release/eucalyptus.1"
+            instance.save()
+            spawn_appserver(instance.ref.pk, mark_active_on_success=True, num_attempts=1)
+
+    @staticmethod
+    def clean_up_after_upgrade(instances):
+        """Remove Eucalyptus upgrade flag from the instance configuration."""
+        for instance in instances:
+            instance.refresh_from_db()
+            InstanceUpgrade._replace_extra_config(instance, "\nCOMMON_EUCALYPTUS_UPGRADE: true\n", "")
+            instance.save()
+
+
+class Eucalyptus1toEucalyptus2(InstanceUpgrade):
+    INITIAL_RELEASE = "eucalyptus.1"
+
+    def do_upgrade(self, instances):
+        """Upgrade all OpenEdXInstances in the given iterable to Eucalyptus."""
+        for instance in instances:
+            instance.refresh_from_db()
+            logger.info("Upgrading instance %s to Eucalyptus.2 ...", instance)
+            self._replace_extra_config(instance, "\nCOMMON_EUCALYPTUS_UPGRADE: true\n", "")
+            instance.edx_platform_repository_url = "https://github.com/open-craft/edx-platform"
+            instance.edx_platform_commit = "opencraft-release/eucalyptus.2"
+            instance.configuration_source_repo_url = "https://github.com/open-craft/configuration"
+            instance.configuration_version = "opencraft-release/eucalyptus.2"
+            instance.openedx_release = "open-release/eucalyptus.2"
```

Comment:
@e-kolpakov It would be nice to have a method that takes care of updating repository URLs and branch names, and just call it from the `do_upgrade` method. I'd be fine with adding that in the next iteration though.